Source/WebGPU/WebGPU/Queue.mm

```diff
@@ -278,14 +278,17 @@
     if (!(buffer.usage() & WGPUBufferUsage_CopyDst))
         return false;
 
+    if (buffer.buffer().storageMode != MTLStorageModePrivate && !buffer.buffer().contents)
```

Comment:
I think we also need a !buffer.isDestroyed() check here like so:
```
if (!buffer.isDestroyed() && buffer.buffer().storageMode != MTLStorageModePrivate && !buffer.buffer().contents)
```
It appears this test fails after this change:

https://gpuweb.github.io/cts/standalone/?q=webgpu:api,validation,encoding,cmds,clearBuffer:buffer_state:bufferState=%22destroyed%22